lib/geomancer/geo_json.ex

```diff
@@ -15,9 +16,34 @@ defmodule Geomancer.GeoJson do
             name: nil,
             bbox: []
 
-  @spec new(features :: [Feature.t()], name :: String.t(), bbox :: map()) :: t()
-  def new(features, name, bbox) do
-    bounding = [bbox.xmin, bbox.ymin, bbox.xmax, bbox.ymax]
-    %__MODULE__{features: features, name: name, bbox: bounding}
+  @spec new({:ok, struct()} | {:error, String.t()}) :: t() | {:error, String.t()}
+  def new({:ok, %Geomancer.Shapefile{} = shapefile}) do
+    features = FeatureSet.reduce(shapefile.shp, type: shapefile.type, properties: shapefile.dbf)
+    bbox = [shapefile.bbox.xmin, shapefile.bbox.ymin, shapefile.bbox.xmax, shapefile.bbox.ymax]
+    %__MODULE__{features: features, name: shapefile.name, bbox: bbox}
+  end
+
+  def new({:error, _} = error), do: error
+
+  @impl Geomancer
+  def from(input_path) do
+    case Path.extname(input_path) do
+      ".zip" ->
+        input_path
+        |> Geomancer.Shapefile.read()
+        |> new()
+        |> Jason.encode()
```

Comment:
Seems weird to new up the struct and then immediately Jason encode it.

Reply:
🤔